src/Symfony/Component/Ldap/LdapClient.php

```diff
@@ -54,6 +54,8 @@ public function __construct($host = null, $port = 389, $version = 3, $useSsl = f
         $this->useSsl = (bool) $useSsl;
         $this->useStartTls = (bool) $useStartTls;
         $this->optReferrals = (bool) $optReferrals;
+
+        $this->connect();
```

Comment:
I almost think there should be a config class or an array of options passed on the constructor. That way you could have something like an auto-connect option that could default to true, but could still be changed if someone wanted to.